receiver/prometheusreceiver/internal/prom_to_otlp_test.go

```diff
@@ -0,0 +1,74 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package internal
+
+import (
+	"sort"
+	"testing"
+	"unsafe"
+
+	metricspb "github.com/census-instrumentation/opencensus-proto/gen-go/metrics/v1"
+	"github.com/google/go-cmp/cmp"
+	"google.golang.org/protobuf/testing/protocmp"
+
+	"go.opentelemetry.io/collector/consumer/pdata"
+	otlpresource "go.opentelemetry.io/collector/internal/data/protogen/resource/v1"
+	"go.opentelemetry.io/collector/translator/internaldata"
+)
+
+func TestCreateNodeAndResourceConversion(t *testing.T) {
+	job, instance, scheme := "converter", "ocmetrics", "http"
+	ocNode, ocResource := createNodeAndResource(job, instance, scheme)
+	mdFromOC := internaldata.OCToMetrics(internaldata.MetricsData{
+		Node:     ocNode,
+		Resource: ocResource,
+		// We need to pass in a dummy set of metrics
+		// just to populate and allow for full conversion.
+		Metrics: []*metricspb.Metric{
+			{
+				MetricDescriptor: &metricspb.MetricDescriptor{
+					Name:        "m1",
+					Description: "d1",
+					Unit:        "By",
+				},
+			},
+		},
+	})
+
+	fromOCResource := protoResource(mdFromOC.ResourceMetrics().At(0).Resource())
+	byDirectOTLPResource := protoResource(createNodeAndResourcePdata(job, instance, scheme))
+
+	if diff := cmp.Diff(byDirectOTLPResource, fromOCResource, protocmp.Transform()); diff != "" {
+		t.Fatalf("Resource mismatch: got: - want: +\n%s", diff)
+	}
+}
+
+// Unfortunately pdata doesn't expose a way for us to retrieve the underlying resource,
+// yet we need to compare the resources which are hidden by an unexported value.
+func protoResource(presource pdata.Resource) *otlpresource.Resource {
+	type extract struct {
```

Comment:
Ugh :(

Reply:
This is a no go. We make no guarantees about pdata representation. In fact, we change it once in a while so this will break.

Reply:
pdata is [likely going](https://github.com/open-telemetry/opentelemetry-collector/issues/3104) to a different module very soon (before GA), possibly to a different repo, together with generated Protobuf messages, placed in an internal package of pdata moddle/repo. There won't be an otlpresource package that will be possible to import here.

Reply:
@tigrannajaryan just to restate what the purpose of this PR is, it is for parity of translation, not code that's going to live in here forever. Whenever am done with the translation and we are sure that we can translate from Prometheus->OTLP directly, the test code for parity is going to be thrown away.

Reply:
Alright, Bogdan suggested to me the solution which was to use .Attributes().Sort() hence no more extracting using unsafe. Thanks everyone for the discussion.

Reply:
If checking the parity with the sorting, etc is important, I think it's ok. But we should add traditional unit tests as well now - we always write unit tests for new code. And we know we'll need them later when deleting the OC-based code right?

Reply:
These equivalence tests are essentially the unit tests, but honestly this is going to be quite the uphill task if we try to make both migrations and then conversion tests at the same time. We have a very long chain of tasks to complete, so I'd kindly suggest let's get code in the migration/parity firstly in, otherwise we are going to blow through the May 31st 2021 deadline (it is already mid way through the month)